src/game_actor.cpp

```diff
@@ -86,11 +86,21 @@ Game_Actor::Game_Actor(int actor_id) {
 	SetHp(GetMaxHp());
 	SetSp(GetMaxSp());
 
-	SetEquipment(1, dbActor->initial_equipment.weapon_id);
-	SetEquipment(2, dbActor->initial_equipment.shield_id);
-	SetEquipment(3, dbActor->initial_equipment.armor_id);
-	SetEquipment(4, dbActor->initial_equipment.helmet_id);
-	SetEquipment(5, dbActor->initial_equipment.accessory_id);
+	std::array<int, 5> ids = {{
+		dbActor->initial_equipment.weapon_id & 0xFFFF,
+		dbActor->initial_equipment.shield_id & 0xFFFF,
+		dbActor->initial_equipment.armor_id & 0xFFFF,
+		dbActor->initial_equipment.helmet_id & 0xFFFF,
+		dbActor->initial_equipment.accessory_id & 0xFFFF }};
+	std::replace_if(ids.begin(), ids.end(), [] (const int& item_id) {
+		return lcf::ReaderUtil::GetElement(lcf::Data::items, item_id) == nullptr;
+	}, 0);
+
+	SetEquipment(1, ids[0]);
+	SetEquipment(2, ids[1]);
+	SetEquipment(3, ids[2]);
+	SetEquipment(4, ids[3]);
+	SetEquipment(5, ids[4]);
```

Comment:
And these 5 SetEquipment calls can be changed to a for-loop.

Reply:
Changed.